examples/token/src/lib.rs

```diff
@@ -0,0 +1,324 @@
+//! The current code supports token symbols, names, maximum supplies, and
+//! simple transfers. It DOES NOT YET support authorizations for delegate
+//! spending.
+//!
+//! ## On developer experience
+//!
+//! As stated above, this is code is still ugly. The developer experience will
+//! improve dramatically as the FVM project advances towards Milestone 2.
+//!
+//! The author deliberately didn't use this opportunity to improve the SDK.
+//! Instead, he chose to implement things low-level to create a baseline of
+//! where we are. As we introduce better DX, this actor will evolve and benefit.
+//!
+//! NOTE: There is _some_ pre-existing sugar in the actor/runtime module
+//! that could facilitate things slightly easier here, but this actor does not
+//! use it to avoid dragging in unnecessary dependencies (related to built-in
+//! actors). The upcoming sugar will eventually be provided by the Rust SDK,
+//! and not the built-in actors toolkit.
+//!
+//! ## Boilerplate
+//!
+//! The following responsiblities are taken care of here, but they are
+//! absolutely boilerplate, and should eventually be superseded by clever sugar
+//! in the form of proc macro attributes.
+//!
+//! - State loading.
+//! - State mutations.
+//! - Method dispatch.
+//! - Deserialization of parameters.
+//! - Serialization of return data.
+//!
+//! ## Addressing
+//!
+//! For simplicity, this actor uses ActorIDs in the balances and allowances
+//! HAMT (map) keys. This poses two problems:
+//!
+//! 1. A sender cannot send tokens to an inexistent account actor.
+//! 2. This actor is not reorg-safe (ActorIDs can change in the chain gets
+//!    reorg-ed).
+//!
+//! We _could_ do sophisticated things here to solve for these problems, by
+//! associating balances to class-{1,2,3} addresses only, and rejecting calls
+//! with class-0 (ID addresses).
+//!
+//! However, all of this will change when we introduce [class-4 addresses](https://github.com/filecoin-project/fvm-specs/blob/main/04-evm-mapping.md#proposed-solution-universal-stable-addresses)
+//! (reorg stable universal addresses), so it's not worth the complexity now,
+//! as this actor only exists for illustration purposes.
+//!
+//! ## NOT MEANT FOR PRODUCTION USAGE
+//!
+//! This actor should NEVER be deployed on the Filecoin network. It is purely
+//! meant for illustration and test purposes.
+//!
+
+use cid::multihash::{Code, MultihashDigest};
+use cid::Cid;
+use fvm_sdk as sdk;
+use fvm_sdk::blockstore::Blockstore;
+use fvm_sdk::message::{params_cbor, NO_DATA_BLOCK_ID};
+use fvm_shared::address::Address;
+use fvm_shared::bigint::bigint_ser::{BigIntDe, BigIntSer};
+use fvm_shared::bigint::{bigint_ser, Zero};
+use fvm_shared::blockstore::CborStore;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::encoding::tuple::*;
+use fvm_shared::encoding::{Cbor, RawBytes, DAG_CBOR};
+use fvm_shared::ActorID;
+use ipld_hamt::Hamt;
+
+/// A macro to abort execution by signalling a non-zero exit code.
+macro_rules! abort {
```

Comment:
Agree, these are just basic local macros because I'm not convinced this is what would ultimately go in an SDK in this particular shape. So I'm not inclined to "bless it" at this stage.